src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -258,6 +359,29 @@ protected String getCreateAccountingDocumentReportEmailAddress() {
                 CuFPParameterConstants.CreateAccountingDocumentService.CREATE_ACCT_DOC_REPORT_EMAIL_ADDRESS);
     }
     
+    protected String getDuplicateFileReportEmailAddress() {
+        return parameterService.getParameterValueAsString(KFSConstants.CoreModuleNamespaces.FINANCIAL, 
+                CuFPParameterConstants.CreateAccountingDocumentService.CREATE_ACCOUNTING_DOCUMENT_SERVICE_COMPONENT_NAME, 
+                CuFPParameterConstants.CreateAccountingDocumentService.DUPLICATE_FILE_REPORT_EMAIL_ADDRESS);
+    }
+    
+    protected Optional<String> getReportEmailFromFileIfPossible(String fileName) {
+        try {
+            AccountingXmlDocumentListWrapper accountingXmlDocuments = unmarshalAccountingDocumentFile(fileName);
+            String reportEmail = accountingXmlDocuments.getReportEmail();
+            if (StringUtils.isNotBlank(reportEmail)) {
+                InternetAddress reportInternetAddress = new InternetAddress(reportEmail);
+                reportInternetAddress.validate();
+                return Optional.of(reportEmail);
+            } else {
+                return Optional.empty();
+            }
+        } catch (Exception e) {
+            LOG.error("getReportEmailFromFileIfPossible, Could not obtain and validate report email from file", e);

```

Comment:
I am thinking about the situation where the XML file is fine, and fully loaded the KFS documents, but the report email address is empty or not valid.  I think we need to alert the developers that we received a file without a report email address.

Reply:
The enclosing `getReportEmailFromFileIfPossible()` method is only meant to be used in cases where the file was rejected due to the duplicate-file check. We already have existing code in `CreateAccountingDocumentValidationServiceImpl` that will validate the correctness of the report email, if the file was not rejected by the duplication check. An email validation error at that phase will cause a header-level validation failure in the file, and will cause the error report to be sent to the email specified by `CREATE_ACCT_DOC_REPORT_EMAIL_ADDRESS`.

If you think we need to adjust the existing code so that such email error reports are also sent to a developer EGA, please let me know.

Reply:
that should be fine, thanks